Firestore/Source/Local/FSTLevelDBQueryCache.mm

```diff
@@ -148,6 +163,53 @@ - (void)setLastRemoteSnapshotVersion:(SnapshotVersion)snapshotVersion {
   _db.currentTransaction->Put([FSTLevelDBTargetGlobalKey key], self.metadata);
 }
 
+- (void)enumerateTargetsUsingBlock:(void (^)(FSTQueryData *queryData, BOOL *stop))block {
+  // Enumerate all targets, give their sequence numbers.
+  std::string targetPrefix = [FSTLevelDBTargetKey keyPrefix];
+  auto it = _db.currentTransaction->NewIterator();
+  it->Seek(targetPrefix);
+  BOOL stop = NO;
+  for (; !stop && it->Valid() && absl::StartsWith(it->key(), targetPrefix); it->Next()) {
+    FSTQueryData *target = [self decodedTarget:it->value()];
+    block(target, &stop);
+  }
+}
+
+- (void)enumerateOrphanedDocumentsUsingBlock:
+    (void (^)(FSTDocumentKey *docKey, FSTListenSequenceNumber sequenceNumber, BOOL *stop))block {
+  std::string documentTargetPrefix = [FSTLevelDBDocumentTargetKey keyPrefix];
+  auto it = _db.currentTransaction->NewIterator();
+  it->Seek(documentTargetPrefix);
+  FSTListenSequenceNumber nextToReport = 0;
+  FSTDocumentKey *keyToReport = nil;
```

Comment:
Please don't use `FSTDocumentKey`. You probably want `absl::optional<DocumentKey>`.

Reply:
Done. We don't actually even need the `optional`, it's value is never checked, just read when appropriate.